src/app.js

```diff
@@ -1,48 +1,47 @@
-export default function (options) {
-	var model = options.model
-	var view = options.view
+export default function (app) {
+	var view = app.view || function () {
+		return ""
+	}
 
+	var model
 	var actions = {}
-	var effects = options.effects || {}
-	var reducers = options.reducers || {}
-	var subs = options.subscriptions
-
-	var router = options.router
-
+	var reducers
+	var effects
+	var subscriptions = []
+	var hooks = []
+	var plugins = app.plugins
 	var node
 	var root
 	var batch = []
 
-	var hooks = merge({
-		onAction: Function.prototype,
-		onUpdate: Function.prototype,
-		onError: function (err) {
+	var onError = function (err) {
+		if (!hook("onError", function (fn) {
```

Comment:
wat? no, no. I not meant that. It's no matter how the var is named. The point is to not be inside the if condition

```js
var foobar = function (fn) { fn(err) }
if (!hook("onError", foobar)) {
  throw err
}
```

and that (as currently) are the same

```js
if (!hook("onError", function (fn) { fn(err) })) { // collapsed in one line
  throw err
}
```